tests/pass/libc.rs

```diff
@@ -1,17 +1,142 @@
 // ignore-windows: No libc on Windows
 // compile-flags: -Zmiri-disable-isolation
 
+#![feature(io_error_more)]
 #![feature(rustc_private)]
 
 extern crate libc;
 
-#[cfg(any(target_os = "linux", target_os = "freebsd"))]
 fn tmp() -> std::path::PathBuf {
     std::env::var("MIRI_TEMP")
         .map(std::path::PathBuf::from)
         .unwrap_or_else(|_| std::env::temp_dir())
 }
 
+/// Test allocating variant of `realpath`.
+fn test_posix_realpath_alloc() {
+    use std::ffi::OsString;
+    use std::ffi::{CStr, CString};
+    use std::fs::{remove_file, File};
+    use std::os::unix::ffi::OsStrExt;
+    use std::os::unix::ffi::OsStringExt;
+    use std::path::PathBuf;
+
+    let buf;
+    let path = tmp().join("miri_test_libc_posix_realpath_alloc");
+    let c_path = CString::new(path.as_os_str().as_bytes()).expect("CString::new failed");
+
+    // Cleanup before test.
+    remove_file(&path).ok();
+    // Create file.
+    drop(File::create(&path).unwrap());
+    unsafe {
+        let r = libc::realpath(c_path.as_ptr(), std::ptr::null_mut());
+        assert!(!r.is_null());
+        buf = CStr::from_ptr(r).to_bytes().to_vec();
+        libc::free(r as *mut _);
+    }
+    let canonical = PathBuf::from(OsString::from_vec(buf));
+    // Cleanup after test.
```

Comment:
This should be properly *after* the test then. ;)

Reply:
I have it here so it still cleans up if the assert fails 🤷

Reply:
None of the other tests do that, and it makes the test a lot harder to read.

That's why we have cleanup *before* the test, so that old junk left around from the previous round does not bother us.